In [1]:
import pandas as pd

In [3]:
titles = pd.read_csv('C:/AI-ML Files/titles.csv')
cast = pd.read_csv('C:/AI-ML Files/cast.csv')

titles.shape[0]
titles.head(5)

title  year
0    Tasveer Mere Sanam  1996
1              Only You  1994
2  El pueblo del terror  1970
3               Machine  2007
4                  MARy  2008

# Q1. How many movies are listed in the titles dataframe?

In [5]:
a = titles.shape[0]

print (a, "movies are listed in the title dataframe.")

244914 movies are listed in the title dataframe.


# Q2. What are the earliest two films listed in the titles dataframe?

In [6]:
x = titles.sort_values(by = 'year')
x.head(2)

title  year
193222          Miss Jerry  1894
51388   The Startled Lover  1898

Miss Jerry and The Startled Lover are the earliest two films listed in the titles dataframe.

# Q3. How many movies have the title 'Hamlet'?

In [12]:
titles[titles['title'] == "Hamlet"].count()

title    20
year     20
dtype: int64

20 movies have the title "Hamlet".

# Q4. How many movies are titled 'North by Northwest'?

In [13]:
titles[titles['title'] == "North by Northwest"].count()

title    1
year     1
dtype: int64

1 movie has the title "North by Northwest".

# Q5. When was the first movie titled "Hamlet" made?

In [14]:
titles[titles['title'] == "Hamlet"].sort_values(by = 'year').head(1)

title  year
52521  Hamlet  1910

The first movie titled "Hamlet" was made in 1910.

# Q6. List all of the "Treasure Island" from the earliest to most recent?

In [15]:
list = titles[titles['title'] == "Treasure Island"].sort_values(by = 'year').values.tolist()

print ('List of all the "Treasure Island" movies from the earliest to most recent -')

x = pd.DataFrame(list, columns =['Title', 'Year'])
x

List of all the "Treasure Island" movies from the earliest to most recent -


Title  Year
0  Treasure Island  1918
1  Treasure Island  1920
2  Treasure Island  1934
3  Treasure Island  1950
4  Treasure Island  1972
5  Treasure Island  1973
6  Treasure Island  1985
7  Treasure Island  1999

# Q7. How many movies were made in the year 1950?

In [18]:
a = titles[titles['year'] == 1950]['title'].count()

print (a, "movies were made in the year 1950.")

1113 movies were made in the year 1950.


# Q8. How many movies were made in the year 1960?

In [19]:
x = titles[titles['year'] == 1960]['title'].count()

print (x, "movies were made in the year 1950.")

1549 movies were made in the year 1950.


# Q9. How many movies were made from 1950 through 1959?

In [21]:
b = titles[(titles['year'] >= 1950) & (titles['year'] <= 1959)]['title'].count()

print (b, "movies were made from 1950 through 1959.")

12934 movies were made from 1950 through 1959.


# Q10. In what years has a movie titled "Batman" been released?

In [33]:
l = titles[titles['title'] == "Batman"]['year'].tolist()
l.sort()
print ("The movie titled 'Batman' has been released in years:\n", *l)

The movie titled 'Batman' has been released in years:
 1943 1989


# Q11. How many roles were there in the movie "Inception"?

In [4]:
import pandas as pd

In [6]:
cast = pd.read_csv('C:/AI-ML Files/cast.csv')

x = cast[cast['title'] == 'Inception']['type'].count()

print (x, "roles were there in the movie inception.")

78 roles were there in the movie inception.


# Q12. How many roles in movie "Inception" are NOT ranked by an "n" value?

In [9]:
p = cast[cast['title'] == 'Inception']['n'].isnull().sum()

print (p, "roles in the movie 'Inception' are NOT ranked by an 'n' value.")

27 roles in the movie 'Inception' are NOT ranked by an 'n' value.


# Q13. But how many roles in the movie "Inception" did receive an "n" value?

In [4]:
import pandas as pd

In [6]:
cast = pd.read_csv('C:/AI-ML Files/cast.csv')

a = cast[cast['title'] == 'Inception']['n'].notnull().sum()

print (a, "roles in the movie 'Inception' did not receive an 'n' value.")

51 roles in the movie 'Inception' did not receive an 'n' value.


# Q14. Display the cast of "North by Northwest" in their correct "n"-value order, ignoring roles that did not earn a numeric "n" value.

In [12]:
a = cast[cast.n.notnull()].sort_values(by=['n'])

l = a[a['title'] == 'North by Northwest'][['name','n']].values.tolist()

print("The cast of 'North by Northwest' in their correct 'n'-value order,  ignoring roles that did not earn a numeric 'n' value-")

n = pd.DataFrame(l, columns = ['Name', 'n']) 
n

The cast of 'North by Northwest' in their correct 'n'-value order,  ignoring roles that did not earn a numeric 'n' value-


Name     n
0             Cary Grant   1.0
1        Eva Marie Saint   2.0
2            James Mason   3.0
3    Jessie Royce Landis   4.0
4         Leo G. Carroll   5.0
5   Josephine Hutchinson   6.0
6            Philip Ober   7.0
7          Martin Landau   8.0
8          Adam Williams   9.0
9           Edward Platt  10.0
10     Robert Ellenstein  11.0
11          Les Tremayne  12.0
12       Philip Coolidge  13.0
13         Patrick McVey  14.0
14          Edward Binns  15.0
15             Ken Lynch  16.0

# Q15. Display the entire cast, in "n"-order, of the 1972 film "Sleuth"?

In [13]:
b = cast[cast.n.notnull()].sort_values(by=['n'])

print('The entire cast, in "n"-order, of the 1972 film "Sleuth"-')

b = b.loc[(b['title']=='Sleuth') & (b['year']==1972)][['name','n']].reset_index(drop=True)
b

The entire cast, in "n"-order, of the 1972 film "Sleuth"-


name    n
0    Laurence Olivier  1.0
1       Michael Caine  2.0
2      Alec Cawthorne  3.0
3  John (II) Matthews  4.0
4  Eve (III) Channing  5.0
5        Teddy Martin  6.0

# Q16. Now display the entire cast, in "n"-order of the 2007 version of "Sleuth"?

In [14]:
c = cast[cast.n.notnull()].sort_values(by=['n'])

print('The entire cast, in "n"-order, of the 2007 film "Sleuth"-')

c = c.loc[(c['title']=='Sleuth') & (c['year']==2007)][['name','n']].reset_index(drop=True)
c

The entire cast, in "n"-order, of the 2007 film "Sleuth"-


name    n
0  Michael Caine  1.0
1       Jude Law  2.0
2  Harold Pinter  3.0

# Q17. How many roles were credited in the silent 1921 version of Hamlet?

In [17]:
d = cast.loc[(cast['title']=='Hamlet') & (cast['year']==1921)]
d['character'].count()

9

9 roles were credited in the silent 1921 version of Hamlet.

# Q18. How many roles were credited in Branagh’s 1996 Hamlet?


In [18]:
e = cast.loc[(cast['title']=='Hamlet') & (cast['year']==1996)]
e['character'].count()

54

54 roles were credited in Branagh's 1996 Hamlet.

# Q19. How many "Hamlet" roles have been listed in all film credits through history?

In [19]:
f = cast[cast['title'] == 'Hamlet']
f['character'].count()

342

342 roles have been listed in all film credits through credits.

# Q20. How many people have played an "Ophelia"?

In [22]:
g = cast[cast['character'] == 'Ophelia']['name'].count()

print (g, "people have played an 'Ophelia'.")

117 people have played an 'Ophelia'.


# Q21. How many people have played a role called "The Dude"?

In [23]:
h = cast[cast['character'] == 'The Dude']['name'].count()

print (h, "people have played a role called 'The Dude'.")

19 people have played a role called 'The Dude'.


# Q22. How many people have played a role called "The Stranger"?

In [24]:
i = cast[cast['character'] == 'The Stranger']['name'].count()

print (i, "people have played a role called 'The Dude'.")

212 people have played a role called 'The Dude'.


# Q23. How many roles has Sidney Poitier played throughout his career?

In [28]:
j = cast[cast['name'] == "Sidney Poitier"]['character'].count()

print ("Sidney Poitier has played", j, "roles throughout his career.")

Sidney Poitier has played 43 roles throughout his career.


# Q24. How many roles has Judi Dench played?

In [29]:
k = cast[cast['name'] == "Judi Dench"]['character'].count()

print ("Judi Dench has played", k, "roles.")

Judi Dench has played 55 roles.


# Q25. List the supporting roles (having n=2) played by Cary Grant in the 1940s, in order by year?

In [30]:
l = cast[(cast['name'] == "Cary Grant") & (cast['n'] == 2) & (cast['year'] >= 1940) & (cast['year'] < 1950)]

li = l[['character','year']].values.tolist()

print("Following are the supporting roles played by Cary Grant in the 1940s-\n")

cg = pd.DataFrame(li, columns = ['Supporting Role', 'Year'])
cg

Following are the supporting roles played by Cary Grant in the 1940s-



Supporting Role  Year
0      Nick Arden  1940
1     Roger Adams  1941

# Q26. List the leading roles that Cary Grant played in the 1940s in order by year?

In [31]:
m = cast[(cast['name'] == "Cary Grant") & (cast['n'] == 1) & (cast['year'] >= 1940) & (cast['year'] < 1950)].sort_values(by = ['year'])

lis= m[['character','year']].values.tolist()

print("Following are the leading roles played by Cary Grant in the 1940s:\n")

cg_lead = pd.DataFrame(lis, columns = ['Leading Role', 'Year'])
cg_lead

Following are the leading roles played by Cary Grant in the 1940s:



Leading Role  Year
0                            Matt Howard  1940
1                           Walter Burns  1940
2                     C. K. Dexter Haven  1940
3                       Johnnie Aysgarth  1941
4                           Leopold Dilg  1942
5                  Patrick 'Pat' O'Toole  1942
6                          Capt. Cassidy  1943
7   Joe Adams -posing as Joe Bascopolous  1943
8                            Jerry Flynn  1944
9                      Mortimer Brewster  1944
10                            Ernie Mott  1944
11                           Cole Porter  1946
12                                Devlin  1946
13                           Dick Nugent  1947
14                                Dudley  1947
15                         Jim Blandings  1948
16                     Dr. Madison Brown  1948
17                 Captain Henri Rochard  1949

# Q27. How many roles were available for actors in the 1950s?

In [32]:
n = cast[(cast['year'] >= 1950) & (cast['year'] < 1960) & (cast['type'] == 'actor')]['character'].count()

print (n, "roles were available for actors in 1950s.")

157738 roles were available for actors in 1950s.


# Q28. How many roles were available for actresses in the 1950s?

In [33]:
o = cast[(cast['year'] >= 1950) & (cast['year'] < 1960) & (cast['type'] == 'actress')]['character'].count()

print (o, "roles were available for actresses in 1950s.")

57733 roles were available for actresses in 1950s.


# Q29. How many leading roles (n=1) were available from the beginning of film history through 1980?

In [34]:
p = cast[(cast['year'] >= 1894) & (cast['year'] <= 1980) & (cast['n'] == 1)]['character'].count()

print (p, "roles were available from beginning of film history through 1980.")

65140 roles were available from beginning of film history through 1980.


# Q30. How many non-leading roles were available through from the beginning of film history through 1980?

In [35]:
q = cast[(cast['year'] >= 1894) & (cast['year'] <= 1980) & (cast['n'] > 1)]['character'].count()

print (q, "non-leading roles were available through the beginning of film history through 1980.")

669320 non-leading roles were available through the beginning of film history through 1980.


# Q31. How many roles through 1980 were minor enough that they did not warrant a numeric "n" rank?

In [36]:
r = cast[(cast['year'] >= 1894) & (cast['year'] <= 1980)]['n'].isnull().sum()

print (r, "roles through 1980 were minor enough that they did not warrant numeric 'n' rank.")

448347 roles through 1980 were minor enough that they did not warrant numeric 'n' rank.
